api/handlers.go

```diff
@@ -41,9 +52,47 @@ type (
 
 // healthGET returns the status of the service
 func (api *API) healthGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	// The public status that we'll return as response to this call.
+	status := &HealthGET{
+		DBAlive: true,
+	}
+	// Extended health status that we'll log for the benefit of the service's
+	// administrators.
+	extHealth := ExtendedHealth{
+		Health:                   status,
+		NumberSessionsInProgress: api.staticDB.NumberSessionsInProgress(),
+	}
+	// Ensure that we log the extended health information after we gather as
+	// much of it as possible.
+	defer func() {
+		b, err := json.Marshal(extHealth)
+		if err != nil {
+			api.staticLogger.Warnf("Failed to serialize extended health information. Error: %v", err)
+		}
+		api.staticLogger.Info(string(b))
+	}()
+	err := api.staticDB.Ping(req.Context())
+	if err != nil {
+		status.DBAlive = false
+		status.Error = err.Error()
+		api.WriteJSON(w, status)
+		return
+	}
+	hello, err := api.staticDB.Hello(req.Context())
+	if err != nil {
+		status.Error = err.Error()
+		api.WriteJSON(w, status)
+		return
+	}
+	extHealth.Hello = hello
+
 	mp, err := conf.MinPinners(req.Context(), api.staticDB)
-	var status HealthGET
-	status.DBAlive = err == nil
+	if err != nil {
+		status.Error = err.Error()
+		status.Primary = hello.Primary
```

Comment:
Knowing the primary is definitely useful. It's also very useful to attackers because it tells them where to focus their efforts. That's why we decided to limit the reporting of the primary to error states only. The primary is always logged, so an operator will be able to get that info from the logs.